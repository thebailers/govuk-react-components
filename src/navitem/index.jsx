import React from 'react';
import PropTypes from 'prop-types';

export const navItemOf = (Type) => {
  const component = (props) => {
    const {active, className = '', ...other} = props;
    const elemClass = (active ? 'active' : '') + (className ? ' ' + className : '');
    return (
      <li>
        <Type {...other} className={elemClass}>{props.children}</Type>
      </li>
    )
  };
  component.defaultName = 'NavItemOf' + Type;
  component.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
    active: PropTypes.bool
  };
  return component;
};

export const NavItem = navItemOf('a');
NavItem.propTypes = Object.assign({ href: PropTypes.string.isRequired }, NavItem.propTypes);

export default NavItem;
