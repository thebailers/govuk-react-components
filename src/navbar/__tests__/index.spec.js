import React from 'react';
import { render } from 'enzyme';
import NavBar from 'govuk-react-components/navbar';
import {NavItem} from 'govuk-react-components/navitem';

describe('The Nav Bar Component', () => {
  it('should render correctly when not passed a propositionHeader link', () => {
    expect(
      render(<NavBar propositionHeader='Lorem ipsum' />)
    ).toMatchSnapshot();
  });

  it('should render correctly when passed a link to the propositionHeader', () => {
    expect(
      render(<NavBar propositionHeader='Lorem ipsum' propositionHeaderLink='/' />)
    ).toMatchSnapshot();
  });

  describe('With navitems', () => {
    it('should render a navitem inside a navbar', () => {
      expect(
        render(
          <NavBar>
            <NavItem href='url-to-page-1' active>Navigation item #1</NavItem>
          </NavBar>
        )
      ).toMatchSnapshot();
    });
  
    it('should render multiple navitems inside a navbar', () => {
      expect(
        render(
          <NavBar>
            <NavItem href='url-to-page-1' active>Navigation item #1</NavItem>
            <NavItem href='url-to-page-2' active>Navigation item #2</NavItem>
            <NavItem href='url-to-page-3' active>Navigation item #3</NavItem>
          </NavBar>
        )
      ).toMatchSnapshot();
    });
  })
});