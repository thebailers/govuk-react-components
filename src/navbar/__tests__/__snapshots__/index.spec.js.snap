// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Nav Bar Component With navitems should render a navitem inside a navbar 1`] = `
<div
  class="header-proposition"
>
  <div
    class="content"
  >
    <nav
      id="proposition-menu"
    >
      <a
        href="/"
        id="proposition-name"
      />
      <ul
        id="proposition-links"
      >
        <li>
          <a
            class="active"
            href="url-to-page-1"
          >
            Navigation item #1
          </a>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;

exports[`The Nav Bar Component With navitems should render multiple navitems inside a navbar 1`] = `
<div
  class="header-proposition"
>
  <div
    class="content"
  >
    <nav
      id="proposition-menu"
    >
      <a
        href="/"
        id="proposition-name"
      />
      <ul
        id="proposition-links"
      >
        <li>
          <a
            class="active"
            href="url-to-page-1"
          >
            Navigation item #1
          </a>
        </li>
        <li>
          <a
            class="active"
            href="url-to-page-2"
          >
            Navigation item #2
          </a>
        </li>
        <li>
          <a
            class="active"
            href="url-to-page-3"
          >
            Navigation item #3
          </a>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;

exports[`The Nav Bar Component should render correctly when not passed a propositionHeader link 1`] = `
<div
  class="header-proposition"
>
  <div
    class="content"
  >
    <nav
      id="proposition-menu"
    >
      <a
        href="/"
        id="proposition-name"
      >
        Lorem ipsum
      </a>
    </nav>
  </div>
</div>
`;

exports[`The Nav Bar Component should render correctly when passed a link to the propositionHeader 1`] = `
<div
  class="header-proposition"
>
  <div
    class="content"
  >
    <nav
      id="proposition-menu"
    >
      <a
        href="/"
        id="proposition-name"
      >
        Lorem ipsum
      </a>
    </nav>
  </div>
</div>
`;
